map.go

```diff
@@ -334,6 +334,9 @@ func newMapWithOptions(spec *MapSpec, opts MapOptions) (_ *Map, err error) {
 // createMap validates the spec's properties and creates the map in the kernel
 // using the given opts. It does not populate or freeze the map.
 func (spec *MapSpec) createMap(inner *sys.FD, opts MapOptions) (_ *Map, err error) {
+	if spec.Type == UnspecifiedMap {
```

Comment:
Could you change this to happen after the map create call has failed? Basically `if err == EINVAL && spec.Type == Unspecified`. That way we avoid validating kernel arguments in userspace.